app/models/audit/event/check.rb

```diff
@@ -14,11 +14,12 @@ def initialize(
       )
         @user = user
         @client_ip = client_ip
-        @resource = resource
         @privilege = privilege
         @role = role
         @operation = operation
         @success = success
+        @resource_id = resource.try(:id) || resource
+        @role_id = role.try(:id) || role
       end
```

Comment:
What do `resource_id` and `role_id` hold? only the id or the entire object that includes them?

Reply:
Both of them hold just the Ids. `@role` holds the entire object and of the role, and about the entire object of resource there is no need in this file for anything but the id property